test/resource/file-format/file-format-1.zap

```diff
@@ -0,0 +1,83 @@
+{
+  "fileFormat": 1,
+  "featureLevel": 96,
+  "creator": "zap",
+  "keyValuePairs": [
+    "commandDiscovery = 1",
+    "defaultResponsePolicy = always",
+    "manufacturerCodes = 0x1002"
+  ],
+  "package": [
+    {
+      "pathRelativity": "relativeToZap",
+      "path": "../../../zcl-builtin/silabs/zcl.json",
+      "type": "zcl-properties",
+      "category": "zigbee",
+      "version": 1,
+      "description": "ZigbeePro test data"
+    }
+  ],
+  "endpointTypes": [
+    {
+      "name": "Endpoint Type A",
+      "deviceTypeName": "TA-billingunit",
+      "deviceTypeCode": 515,
+      "deviceTypeProfileId": 263,
+      "clusters": [
+        {
+          "name": "Basic",
+          "code": "0x0000",
+          "define": "BASIC_CLUSTER",
+          "side": "server",
+          "enabled": 1,
+          "attributes": [
+            "+ | 0x0000 |        | server | RAM |           |       |                 0x08 | 0 | 0 | 65534 | 0 => ZCL version [int8u]",
```

Comment:
So the main change here is that attributes, instead of being stored as actual objects in the JSON, are now being stored as some sort of string that ZAP will parse?

This seems like a significant step backwards to me.  It includes even more information in the .zap file that duplicates the XML (e.g. the attribute types) in a way that makes it even harder to remove.  And there's no obvious documentation for the fields here, which makes working with the .zap files without using the ZAP UI (a common ask) more complicated.

Reply:
You are correct that gzipping the JSON would be a much more saving. However this is attempting to find a ballance between:
  - make files smaller
AND
  - keep them in text format, readable and writable by humans in text editor.
  
  The main problem is, that storing an attribute takes 16 lines of text:
  ```            
  {
              "name": "ClusterRevision",
              "code": 65533,
              "mfgCode": null,
              "side": "client",
              "type": "int16u",
              "included": 1,
              "storageOption": "RAM",
              "singleton": 0,
              "bounded": 0,
              "defaultValue": "2",
              "reportable": 1,
              "minInterval": 1,
              "maxInterval": 65534,
              "reportableChange": 0
}
```
So if you go scrolling over the text file with 6 blocks like this, you get lost in this.
A block of few attributes like this, is arguably easier to peruse:
```          
"attributes": [
            "+ | 0x0000 |        | server | RAM |           |       |                 0x00 | 1 | 0 | 65344 | 0 => SceneCount [int8u]",
            "+ | 0x0001 |        | server | RAM |           |       |                 0x00 | 1 | 0 | 65344 | 0 => CurrentScene [int8u]",
            "+ | 0x0002 |        | server | RAM |           |       |               0x0000 | 1 | 0 | 65344 | 0 => CurrentGroup [group_id]",
            "+ | 0x0003 |        | server | RAM |           |       |                 0x00 | 1 | 0 | 65344 | 0 => SceneValid [boolean]",
            "+ | 0x0004 |        | server | RAM |           |       |                      | 1 | 0 | 65344 | 0 => NameSupport [bitmap8]",
            "+ | 0xfffd |        | server | RAM |           |       |                    4 | 1 | 0 | 65344 | 0 => ClusterRevision [int16u]"
]
```

But you are correct, that this now requires custom parsing, so this is 100% detrimental to scripted post-processing of zap files. Zap, though assumes that you are doing all the .zap file processing through zap itself. But obviously, that's a "wish" and possibly not a reality.

Note that there is NO MORE information, it's just differently formatted. We already had types and names there that served nothing.

I dunno.... I am not going to merge this in as a default file format, if you are strongly unhappy about it. 

I can add file-format 2, which would be gzipped files?

Any other idea?

Reply:
Honestly, doing gunzip, edit, gzip (or unzip, edit, zip) is a lot simpler than figuring out the custom syntax even for hand-editing...

And yes, there are already tools out there that do .zap file processing that are not ZAP itself.  I guess those could be supported if there were a way to take the new format, convert to old, modify, convert to new.

> Note that there is NO MORE information, it's just differently formatted. 

Right, but it seems like it makes it harder to _remove_ the redundant information that shouldn't really be there, if we wanted to do that in the future, since it bakes that information into the syntax.

Reply:
That sounds great, thank you!